hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/DumpReplicationQueues.java

```diff
@@ -294,80 +327,99 @@ public String dumpPeersState(List<ReplicationPeerDescription> peers) throws Exce
-    // sb.append(formatQueue(regionserver, queueStorage, queueInfo, queueId, wals, true, hdfs));
-    // } else {
-    // sb.append(formatQueue(regionserver, queueStorage, queueInfo, queueId, wals, false, hdfs));
-    // }
-    // }
-    // }
+      if (!liveRegionServers.contains(regionServer)) {
+        deadRegionServers.add(regionServer.getServerName());
+      }
+      for (ReplicationQueueId queueId : queueIds) {
+        List<String> wals = null;
+        if (queueId.isRecovered()) {
+          wals = AbstractFSWALProvider
+            .getArchivedWALFiles(connection.getConfiguration(), queueId.getSourceServerName().get(),
+              URLEncoder.encode(queueId.getSourceServerName().get().toString(),
+                StandardCharsets.UTF_8.name()))
+            .stream().map(Path::toString).collect(Collectors.toList());
+        } else {
+          wals =
+            AbstractFSWALProvider
+              .getArchivedWALFiles(connection.getConfiguration(), queueId.getServerName(),
+                URLEncoder.encode(queueId.getServerName().toString(),
+                  StandardCharsets.UTF_8.name()))
+              .stream().map(Path::toString).collect(Collectors.toList());
+        }
+        Collections.sort(wals);
+        if (!peerIds.contains(queueId.getPeerId())) {
+          deletedQueues.add(regionServer + "/" + queueId);
+          sb.append(formatQueue(regionServer, queueStorage, wals, queueId, true, hdfs));
+        } else {
+          sb.append(formatQueue(regionServer, queueStorage, wals, queueId, false, hdfs));
+        }
+      }
+    }
     return sb.toString();
   }
 
-  private String formatQueue(ServerName regionserver, ReplicationQueueStorage queueStorage,
-    ReplicationQueueInfo queueInfo, String queueId, List<String> wals, boolean isDeleted,
-    boolean hdfs) throws Exception {
+  private String formatQueue(ServerName regionServer, ReplicationQueueStorage queueStorage,
+    List<String> wals, ReplicationQueueId queueId, boolean isDeleted, boolean hdfs)
+    throws Exception {
     StringBuilder sb = new StringBuilder();
 
-    List<ServerName> deadServers;
-
-    sb.append("Dumping replication queue info for RegionServer: [" + regionserver + "]" + "\n");
+    sb.append("Dumping replication queue info for RegionServer: [" + regionServer + "]" + "\n");
     sb.append("    Queue znode: " + queueId + "\n");
-    sb.append("    PeerID: " + queueInfo.getPeerId() + "\n");
-    sb.append("    Recovered: " + queueInfo.isQueueRecovered() + "\n");
-    deadServers = queueInfo.getDeadRegionServers();
-    if (deadServers.isEmpty()) {
-      sb.append("    No dead RegionServers found in this queue." + "\n");
+    sb.append("    PeerID: " + queueId.getPeerId() + "\n");
+    sb.append("    Recovered: " + queueId.isRecovered() + "\n");
+    if (queueId.getSourceServerName().isPresent()) {
+      sb.append("    Dead RegionServers: " + queueId.getSourceServerName().get() + "\n");
```

Comment:
Actually here we only record the first region server instead of the all the region servers, so maybe we should change the wording here in the future...